### PR TITLE
Alex/docs overhaul

### PR DESCRIPTION
**Pull request by alex-bezek**
related: #165

Explore the docs here: https://github.com/ngrok/kubernetes-ingress-controller/blob/alex/docs-overhaul/docs/README.md

## What

Create a set of doc pages in the readme that eventually will become part of the doc site. This content is broken into sections linked from the main readme:
- deployment-guide: guides for installing the controller and configuration settings at the controller level
- developer-guide: documentation on internals such as the architecture, cds, and releasing
- examples: this isn't really filled out, but will house a few basic examples we can reference
- user-guide: guides for people creating ingress objects trying to manage ingress to their services. 

There are a number of spots that need either a bit more work or aren't filled out at all. For now, those have the Construction image on them as a TODO that we can follow up with.

### CODE REVIEW

File: docs/developer-guide/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Developer Guide
+
+While this project is new and we don't have full contribution guidelines yet, we are open to any PRs and or issues raised. Feel free to reach out to us on slack as well!
```
Comment (jonstacks) [inline docs/developer-guide/README.md:3]: *Non-Blocking*: We should make a `CONTRIBUTING.md` and add it to the root.

File: docs/developer-guide/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## In the weeds architecture
+
+Have a look at the architecture guide on the internal workings of the ingress controller [here](./architecture.md)
+
+## Local Development
+
+* [Go 1.19](https://go.dev/dl/)
```
Comment (jonstacks) [inline docs/developer-guide/README.md:11]: We are on `1.20` now.

File: docs/user-guide/ingress-to-edge-relationship.md [hunk 3]
```diff
@@ -135,31 +135,46 @@ This configuration would produce two edges with one route each.
...
 apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
   name: example-header-add-1
   annotations:
-    k8s.ngrok.com/response-headers-add: |
-      {
-        "X-SEND-TO-CLIENT": "Value1"
-      }
+    k8s.ngrok.com/modules=example-module-set-1
```
Comment (jonstacks) [inline docs/user-guide/ingress-to-edge-relationship.md:177]:
```suggestion
    k8s.ngrok.com/modules: "example-module-set-1"
```

File: docs/user-guide/ingress-to-edge-relationship.md [hunk 4]
```diff
@@ -178,10 +193,7 @@ kind: Ingress
 metadata:
   name: example-header-add-2
   annotations:
-    k8s.ngrok.com/response-headers-add: |
-      {
-        "X-SEND-TO-CLIENT": "Value2"
-      }
+    k8s.ngrok.com/modules=example-module-set-2
```
Comment (jonstacks) [inline docs/user-guide/ingress-to-edge-relationship.md:196]:
```suggestion
    k8s.ngrok.com/modules: "example-module-set-2"
```

**Review by jonstacks — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -51,93 +51,9 @@ An official roadmap is coming soon. In the meantime, here are some of the featur
 * Free tier support
```
Comment (alex-bezek) [inline README.md:51]: Oh, I guess I should clean up the rest of the main readme. I'll clean these up

**Pull request merged**